src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/ApplicationServiceTest.kt

```diff
@@ -693,6 +763,56 @@ class ApplicationServiceTest {
 
     verify { mockApplicationRepository.save(any()) }
     verify(exactly = 1) { mockAssessmentService.createAssessment(application) }
+
+    verify(exactly = 1) {
+      mockDomainEventService.save(
+        match {
+          val data = (it.data as ApplicationSubmittedEnvelope).eventDetails
+          val firstTeam = staffUserDetails.teams!!.first()
+
+          it.applicationId == application.id &&
+            it.crn == application.crn &&
+            data.applicationId == application.id &&
+            data.applicationUrl == "http://frontend/applications/${application.id}" &&
+            data.personReference == PersonReference(
+            crn = offenderDetails.otherIds.crn,
+            noms = offenderDetails.otherIds.nomsNumber!!
+          ) &&
+            data.deliusEventNumber == application.eventNumber &&
+            data.releaseType == "release-type" &&
+            data.age == Period.between(offenderDetails.dateOfBirth, LocalDate.now()).years &&
+            data.gender == ApplicationSubmitted.Gender.male &&
+            data.submittedBy == ApplicationSubmittedSubmittedBy(
+            staffMember = StaffMember(
+              staffCode = staffUserDetails.staffCode,
+              staffIdentifier = staffUserDetails.staffIdentifier,
+              forenames = staffUserDetails.staff.forenames,
+              surname = staffUserDetails.staff.surname,
+              username = staffUserDetails.username
+            ),
+            probationArea = ProbationArea(
+              code = staffUserDetails.probationArea.code,
+              name = staffUserDetails.probationArea.description
+            ),
+            team = Team(
+              code = firstTeam.code,
+              name = firstTeam.description
+            ),
+            ldu = Ldu(
+              code = firstTeam.teamType.code,
+              name = firstTeam.teamType.description
+            ),
+            region = Region(
+              code = staffUserDetails.probationArea.code,
+              name = staffUserDetails.probationArea.description
+            )
+          ) &&
+            data.mappa == risks.mappa.value!!.level &&
+            data.sentenceLengthInMonths == null &&
+            data.offenceId == application.offenceId
+        }
+      )
+    }
```

Comment:
Bravo for this epic piece of mocking! 

I think building the event requires more API calls than it should because we are having to obtain some pieces of information for our Domain Event which ought to be part of the data/document -- as they are also needed for displaying the read-only "point-in-time" application:

- Community API: `offenderService.getRiskByCrn`:
	- mappa value 

- Community API: `offenderService.getOffenderByCrn`:
	- age
	- NOMS number
	- gender

- sentence length (not currently displayed)

Perhaps the right approach to fixing this deficiency would be to 

1. have the calling `ApplicationService.submitApplication()` function make these API calls to Community API and
2.  apply the "missing" first class attributes to the application before 
3.  passing it through to this `createApplicationSubmittedEvent()` function

Reply:
As discussed on Slack - will leave as-is for now as this info should probably be placed into the `data` blob by the frontend and displayed on check your answers.